beacon-chain/rpc/eth/beacon/blocks.go

```diff
@@ -201,61 +202,57 @@ func (bs *Server) SubmitBlock(ctx context.Context, req *ethpbv2.SignedBeaconBloc
-	var err error
-	var v1alpha1Blk interface{}
-
-	switch blk := req.Message.(type) {
-	case *ethpbv2.SignedBeaconBlockContainerV2_Phase0Block:
-		v1alpha1Blk, err = migration.V1ToV1Alpha1SignedBlock(&ethpbv1.SignedBeaconBlock{Block: blk.Phase0Block, Signature: req.Signature})
-		if err != nil {
-			return nil, status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block: %v", err)
+	phase0BlkContainer, ok := req.Message.(*ethpbv2.SignedBeaconBlockContainerV2_Phase0Block)
+	if ok {
+		if err := bs.submitPhase0Block(ctx, phase0BlkContainer.Phase0Block, req.Signature); err != nil {
+			return nil, err
 		}
-		wsb, err = wrapper.WrappedSignedBeaconBlock(v1alpha1Blk)
-	case *ethpbv2.SignedBeaconBlockContainerV2_AltairBlock:
-		v1alpha1Blk, err = migration.AltairToV1Alpha1SignedBlock(&ethpbv2.SignedBeaconBlockAltair{Message: blk.AltairBlock, Signature: req.Signature})
-		if err != nil {
-			return nil, status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block: %v", err)
+	}
+	altairBlkContainer, ok := req.Message.(*ethpbv2.SignedBeaconBlockContainerV2_AltairBlock)
+	if ok {
+		if err := bs.submitAltairBlock(ctx, altairBlkContainer.AltairBlock, req.Signature); err != nil {
+			return nil, err
 		}
-		wsb, err = wrapper.WrappedSignedBeaconBlock(v1alpha1Blk)
-	case *ethpbv2.SignedBeaconBlockContainerV2_BellatrixBlock:
-		v1alpha1Blk, err = migration.BellatrixToV1Alpha1SignedBlock(&ethpbv2.SignedBeaconBlockBellatrix{Message: blk.BellatrixBlock, Signature: req.Signature})
-		if err != nil {
-			return nil, status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block: %v", err)
+	}
+	bellatrixBlkContainer, ok := req.Message.(*ethpbv2.SignedBeaconBlockContainerV2_BellatrixBlock)
+	if ok {
+		if err := bs.submitBellatrixBlock(ctx, bellatrixBlkContainer.BellatrixBlock, req.Signature); err != nil {
+			return nil, err
 		}
-		wsb, err = wrapper.WrappedSignedBeaconBlock(v1alpha1Blk)
-	default:
-		return nil, status.Errorf(codes.InvalidArgument, "Unsupported block type %T", req.Message)
 	}
+	return &emptypb.Empty{}, nil
+}
 
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "Could not wrap beacon block: %v", err)
-	}
+// SubmitBlindedBlock instructs the beacon node to use the components of the `SignedBlindedBeaconBlock` to construct
+// and publish a `SignedBeaconBlock` by swapping out the `transactions_root` for the corresponding full list of `transactions`.
+// The beacon node should broadcast a newly constructed `SignedBeaconBlock` to the beacon network,
+// to be included in the beacon chain. The beacon node is not required to validate the signed
+// `BeaconBlock`, and a successful response (20X) only indicates that the broadcast has been
+// successful. The beacon node is expected to integrate the new block into its state, and
+// therefore validate the block internally, however blocks which fail the validation are still
+// broadcast but a different status code is returned (202).
```

Comment:
Doesn't 20X cover 202?

Reply:
I copied this description from the spec. I think what this tries to say is that a code different than `200` is returned.